framework/src/base/Assembly.C

```diff
@@ -1547,3 +1547,9 @@ Assembly::clearCachedJacobianContributions()
   _cached_jacobian_contribution_cols.reserve(1.2*orig_size);
   _cached_jacobian_contribution_vals.reserve(1.2*orig_size);
 }
+
+bool
+Assembly::isBlockDiagonal()
```

Comment:
I don't understand why this was added - it doesn't seem to be getting called anywhere in your PR....